fix: add back panel prop to IrisGrid Plugin

AkshatJawne opened the pull request:
Closes #2093  (For now, will reopen ticket after merge for better solution to handle deprecated `panel` prop)

mofojed commented on packages/dashboard-core-plugins/src/panels/IrisGridPanel.tsx:
```diff
@@ -423,6 +423,11 @@ export class IrisGridPanel extends PureComponent<
         return null;
       }
 
+      // TODO: Find a better way to handle deprecated panel prop
```
For a TODO like this, we should reference a ticket number and have a ticket created. We can clone the current ticket, or just use the same ticket since this is kind of an interim solution. We want something better.

```suggestion
      // TODO #2096: Find a better way to handle deprecated panel prop
```

AkshatJawne commented:
Added current ticket number in TODO comment, and updated description to indicate that ticket will reopen after this merge to look at a more long-term, proper solution.
